### PR TITLE
Implement mul for overlay * holomap

### PR DESCRIPTION
**Pull request by jlstevens**
Attempting to fix the issues in #3208 and #3210 - there was a submodule update in those PRs that may have been intentional but may also have been accidental (explaining the issue).

### CODE REVIEW

File: holoviews/core/overlay.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def dynamic_mul(*args, **kwargs):
             return other.clone(shared_data=False, callback=callback,
                                streams=[])
         elif not isinstance(other, ViewableElement):
-            return NotImplemented
+            raise NotImplementedError
```
Comment (jlstevens) [inline holoviews/core/overlay.py]: I remember there is special semantics of `NotImplemented` so now I am wondering if that was the right thing.

Comment (jlstevens) [reply holoviews/core/overlay.py]: I.e:

```suggestion
            return NotImplemented
```

Comment (philippjfr) [reply holoviews/core/overlay.py]: ``return NotImplemented`` in ``__mul__`` force it to use ``__rmul__`` which is important to get the HoloMaps working.

Comment (jlstevens) [reply holoviews/core/overlay.py]: Right, I figured that out and fixed it accordingly. Was I right to remove the submodule reference update? The tests are now green.

Comment (philippjfr) [reply holoviews/core/overlay.py]: Yes, should be ready to merge.

**Pull request merged**